components/server/src/server.ts

```diff
@@ -334,13 +334,13 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
         if (!this.config.runDbDeleter) {
             return;
         }
-        const areWeLeader = await this.qorum.areWeLeader();
-        if (areWeLeader) {
-            log.info("[PeriodicDbDeleter] Current instance is leader, starting periodic deleter.");
-            this.periodicDbDeleter.start();
-        } else {
-            log.info("[PeriodicDbDeleter] Current instance is not the leader, periodic deleter will not run.");
-        }
+        this.periodicDbDeleter.start(async () => {
```

Comment:
periodic deleter lives in gitpod-db, therefore I opted for the callback as signal of enablement.